### PR TITLE
!feat(remix-edge-adapter): update default config

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
> **Warning**  
> This would be a breaking change

This one is necessary for the changes done in https://github.com/netlify/remix-template/pull/115

### CODE REVIEW

**Comment by nickytonline** [comment]
I pulled this locally and I didn't dig too deep @MichaelDeBoey as I have other priorities at the moment, but I had to upgrade to the latest Remix so that it didn't complain about the Remix config.

```
npm upgrade @remix-run/dev @remix-run/eslint-config @remix-run/serve -w=packages/edge-demo-site --save
```

CleanShot 2023-05-15 at 17 37 58

I also had to rename https://github.com/netlify/remix-compute/blob/main/packages/edge-demo-site/server.js to server.ts. 

```
Building Remix app in production mode...

✘ [ERROR] Could not resolve "./server.ts"


Build failed with 1 error:
error: Could not resolve "./server.ts"
Error
    at Object.onCompileFailure (/Users/nicktaylor/dev/work/remix-compute/node_modules/@remix-run/dev/dist/cli/commands.js:193:13)
    at Object.compile (/Users/nicktaylor/dev/work/remix-compute/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:45:134)
    at async Object.build (/Users/nicktaylor/dev/work/remix-compute/node_modules/@remix-run/dev/dist/compiler/build.js:33:3)
    at async Object.build (/Users/nicktaylor/dev/work/remix-compute/node_modules/@remix-run/dev/dist/cli/commands.js:188:3)
    at async Object.run (/Users/nicktaylor/dev/work/remix-compute/node_modules/@remix-run/dev/dist/cli/run.js:456:7)
npm ERR! Lifecycle script `build` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: edge-demo-site@1.0.0 
npm ERR!   at location: /Users/nicktaylor/dev/work/remix-compute/packages/edge-demo-site
```

I got everything built, but then it errors out when running locally at least. Like I said, I haven't dug too deep yet.

CleanShot 2023-05-15 at 17 44 41

And not much in the logs.

```
◈ Static server listening to 3999

   ┌─────────────────────────────────────────────────┐
   │                                                 │
   │   ◈ Server now ready on http://localhost:8888   │
   │                                                 │
   └─────────────────────────────────────────────────┘

◈ Loaded edge function server
[server] undefined
[server] undefined
```

If you make those changes on your local, do you exhibit the same behaviour?

**Comment by nickytonline** [comment]
Do you know if it's a requirement for the server file to be a TypeScript file if the project is a TS project?

**Comment by MichaelDeBoey** [comment]
@nickytonline It's not a requirement, but we do recommend using TS fully in the Remix repo and have changed all `.js` server files to TS in our templates (see https://github.com/remix-run/remix/pull/5822)

**Comment by nickytonline** [comment]
> @nickytonline It's not a requirement, but we do recommend using TS fully in the Remix repo and have changed all `.js` server files to TS in our templates (see [remix-run/remix#5822](https://github.com/remix-run/remix/pull/5822))
> 
> Cypress seems to fail on something unrelated though 🤔

Cypress is failing as we have a GH token that external users do not have access to. I approved it to run, but it still complains. I need to look into that.

**Comment by MichaelDeBoey** [comment]
@nickytonline Anything I still need to do in order to get this one merged?

**Comment by nickytonline** [comment]
Looks like the application errors is due to meta data. When I debug and enable pause on caught exceptions, I get the following error, `throw new Error("The `v2_meta` API is enabled in the Remix config, but the route at " + _match.route.path + ` returns an invalid value. In v2, all route meta functions must return an array of meta objects.`

I'll go ahead and fix that.

**Comment by nickytonline** [comment]
It runs locally now, but there is a build issue that I'll need to look into.

```
4:11:48 PM: Edge Functions bundling                                       
4:11:48 PM: ────────────────────────────────────────────────────────────────
4:11:48 PM: ​
4:11:48 PM: Packaging Edge Functions from .netlify/edge-functions directory:
4:11:48 PM:  - server
4:11:49 PM: TypeError: Cannot set property emit of #<Object> which has only a getter
4:11:49 PM:     at shimEmitUncaughtException (file:///opt/build/repo/.netlify/edge-functions/server.js:11328:28)
4:11:49 PM:     at Object.exports13.install (file:///opt/build/repo/.netlify/edge-functions/server.js:11358:90)
4:11:49 PM:     at ../../node_modules/@remix-run/node/dist/index.js (file:///opt/build/repo/.netlify/edge-functions/server.js:24344:39)
4:11:49 PM:     at file:///opt/build/repo/.netlify/edge-functions/server.js:10:50
4:11:49 PM:     at file:///opt/build/repo/.netlify/edge-functions/server.js:29132:27
4:11:49 PM: ​
4:11:49 PM: Bundling of edge function failed                              
4:11:49 PM: ────────────────────────────────────────────────────────────────
```

https://app.netlify.com/sites/remix-edge/deploys/64711225cab5ef000848dbc7#L101-L114

Looks like you're using `@remix-run/node` a Node.js package that doesn't work in a Deno environment. Also `node:stream` isn't supported atm. Same for `renderToPipeableStream`. We can use Webstreams.

You can usewhat is here:
- https://github.com/netlify/remix-template/discussions/100
- https://github.com/netlify/remix-template/commit/fd3139b9458589df7e31ead8cd353d219452400b

**Comment by MichaelDeBoey** [comment]
@nickytonline I updated `edge-demo-site` to bring in line with latest official Remix templates

**Comment by nickytonline** [comment]
> @nickytonline I updated `edge-demo-site` to bring in line with latest official Remix templates

That works for a Node.js environment but causes our build to fail. You'll need to follow what I mentioned here. https://github.com/netlify/remix-compute/pull/84#issuecomment-1564904445

**Comment by MichaelDeBoey** [comment]
@nickytonline Seems like I copied over the wrong version of `entry.server.tsx`
Should be fine now I think

File: packages/edge-demo-site/app/entry.server.tsx [hunk 1]
```diff
@@ -1,28 +1,27 @@
-import type { EntryContext } from '@remix-run/server-runtime'
+/**
+ * By default, Remix will handle generating the HTTP Response for you.
+ * You are free to delete this file if you'd like to, but if you ever want it revealed again, you can run `npx remix reveal` ✨
+ * For more information, see https://remix.run/file-conventions/entry.server
+ */
+
+import type { AppLoadContext, EntryContext } from '@remix-run/node'
 import { RemixServer } from '@remix-run/react'
-import { renderToReadableStream } from 'react-dom/server'
```
Comment (MichaelDeBoey) [inline packages/edge-demo-site/app/entry.server.tsx:3]: @nickytonline It seems like the original did use `renderToReadableStream` though 🤔

**Comment by MichaelDeBoey** [comment]
@nickytonline Build passes again now 🥳🎉

File: packages/edge-demo-site/app/entry.server.tsx (hunk 1 above)
Comment (nickytonline) [reply packages/edge-demo-site/app/entry.server.tsx:3]: That's correct. I removed it due to some DX issues, but I will most likely be adding it back. The Remix dev server doesn't support custom servers, which is why I went with the non-streaming version that runs in a Node context. It applies less to this repo the DX issue. It's more in the Remix template repo. It's related to the discussion I linked.

**Comment by nickytonline** [comment]
I still have issues running E2E tests for external contributors, but I ran them locally and all good.

CleanShot 2023-05-26 at 17 10 07@2x

**Review by nickytonline — APPROVED** [review]
Thanks for working on this @MichaelDeBoey!

**Comment by MichaelDeBoey** [comment]
@nickytonline I rebased onto latest `main` & fixed conflicts
Is there anything else that needs to be done in order to get this one merged?

**Comment by nickytonline** [comment]
> @nickytonline I rebased onto latest `main` & fixed conflicts Is there anything else that needs to be done in order to get this one merged?

No, I think that's it. I'll look to get this merged this week.

**Comment by MichaelDeBoey** [comment]
Thanks for the merge @nickytonline! 🙏
Any ETA on when this will be released?

**Comment by nickytonline** [comment]
It's typically automated, but I think some processes were updated and it's stopped the auto-pr for releases. I'll need to look into this.

**Pull request merged**